common/app/model/ArticleBlocks.scala

```diff
@@ -34,12 +34,17 @@ case object GenericFallback extends BlockRange {
 
 case class PageWithBlock(page: String) extends BlockRange {
   // just get them all, the caching should prevent too many requests, could use "around"
-  val query = None
+  val timeline = "body:key-events"
+  val summary = "body:summary"
+  val body = "body"
+  val query = Some(Seq(body, timeline, summary))
```

Comment:
I can't quite work out why we need to add these parameters when requesting an older page?

if we do, in theory, because we always request all blocks, timeline and summary blocks should always be available as a subset of that response, and you could filter based on the block type to find them; I guess it's a trade off between increasing the size of the response from capi by adding these query parameters and the convenience of the response having timeline and summary items grouped in the requestedbodyblocks section - maybe the size of the response isn't a huge concern while we don't have many key events and summaries, but it's worth baring in mind

Reply:
I think this is an interesting question but for me it came down to consistency.

In _some_ requests we don't have the whole set of `body` blocks, which means we'd need to sometimes query them from `blocks.requestedBodyBlocks.x` and other times filter them from `blocks.body`.

To me this wasn't very intuitive, and offloading the work to CAPI seemed like it would make sense.

I think this was causing issues already, for example `blocks.body` was always being used for getting the summary blocks for key events in DCR, however blocks.body wasn't always defined ~ I'm guessing this was never noticed when the liveblog DCR model was built.

I also definitely prefer that with this solution it's always CAPIs responsibility to filter the blocks. If at any time there's a change to how these blocks should be filtered, we reduce the chance of platform inconsistencies.

However you are right, this comes at the cost of some duplicated data in the request response, which isn't ideal (though I don't have a good grasp on how much of a difference it makes).